hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerManagerImpl.java

```diff
@@ -60,6 +64,7 @@
   /**
    *
    */
+  //Can we move this lock to ContainerStateManager?
```

Comment:
Same question here. I see we have a RWLock in ContainerMangerImpl and one RWLock in ContainerStateManager. Can we rely only on the lock in ContainerStateManager?

Reply:
I would also like to move the lock to ContainerStateManager and not have any lock in ContainerMangerImpl. I'm still thinking of corner cases where this might cause issues.
State change calls to ContainerStateManager are made via Ratis, so we should not check some state in ContainerMangerImpl and make a Ratis call to change the state without holding any lock.

I will remove the lock in followup jira after making sure that we are not missing any corner cases.

Reply:
Agree with @xiaoyuyao 's suggestion.
@nandakumar131 Can we also track it in a separate Jira?